src/main/java/marquez/api/models/DatasourceRequest.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package marquez.api.models;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import javax.validation.constraints.NotNull;
+import lombok.AllArgsConstructor;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.ToString;
+
+@AllArgsConstructor(onConstructor = @__(@JsonCreator))
+@EqualsAndHashCode
+@ToString
+public class DatasourceRequest {
+  @Getter @NotNull private String connectionUrl;
+  @Getter @NotNull private String name;
```

Comment:
Minor: Though order of fields don't matter when deserializing the request payload, for consistency in how they are / will be defined in `openapi.yml` I'd order them as:

```java
DatasourceRequest {
  name
  connectionUrl
}
```

Reply:
You'll want to use `@NotBlank` instead of `@NotNull`. That is, `@NotBlank`: _a constrained String is valid as long as it’s not null and the trimmed length is greater than zero_. Which is really what we want :)